cudapoa/src/allocate_block.cpp

```diff
@@ -259,13 +265,25 @@ void BatchBlock::get_alignment_details(AlignmentDetails** alignment_details_d_p)
     *alignment_details_d_p                   = alignment_details_d;
 }
 
-void BatchBlock::get_graph_details(GraphDetails** graph_details_d_p)
+void BatchBlock::get_graph_details(GraphDetails** graph_details_d_p, GraphDetails** graph_details_h_p)
 {
     GraphDetails* graph_details_d{};
+    GraphDetails* graph_details_h{};
 
     // on host
+    graph_details_h = reinterpret_cast<GraphDetails*>(&block_data_h_[offset_h_]);
+    offset_h_ += sizeof(GraphDetails);
+    graph_details_h->nodes = &block_data_h_[offset_h_];
+    offset_h_ += sizeof(uint8_t) * max_nodes_per_window_ * max_poas_;
+    graph_details_h->incoming_edges = reinterpret_cast<uint16_t*>(&block_data_h_[offset_h_]);
```

Comment:
This may lead to a problem if `offset_h_` happens to be odd here, because `uint16_t` has stricter alignment requirements than `uint8_t` on most platforms.

Reply:
Fair point. Not likely to happen because our max nodes and all are defined to be even numbers, but I'll try to address this as well

Reply:
Will address in a separate PR - https://github.com/clara-genomics/ClaraGenomicsAnalysis/issues/215